lib/SILOptimizer/FunctionSignatureTransforms/ExistentialTransform.cpp

```diff
@@ -0,0 +1,590 @@
+      /// Determine the Conformances.
+      SmallVector<ProtocolConformanceRef, 1> NewConformances;
+      auto ContextTy = NewF.mapTypeIntoContext(GenericParam);
+      auto OpenedArchetype = ContextTy->castTo<ArchetypeType>();
+      for (auto proto : OpenedArchetype->getConformsTo()) {
+        NewConformances.push_back(ProtocolConformanceRef(proto));
+      }
+      ArrayRef<ProtocolConformanceRef> Conformances =
+          Ctx.AllocateCopy(NewConformances);
+      auto ExistentialRepr =
+          ArgDesc.Arg->getType().getPreferredExistentialRepresentation(M);
+      switch (ExistentialRepr) {
+      case ExistentialRepresentation::Opaque: {
+        /// Create this sequence for init_existential_addr.:
+        /// bb0(%0 : $*T):
+        /// %3 = alloc_stack $P
+        /// %4 = init_existential_addr %3 : $*P, $T
+        /// copy_addr %0 to [initialization] %4 : $*T
+        /// destroy_addr %0 : $*T
+        /// %7 = open_existential_addr immutable_access %3 : $*P to
+        /// $*@opened P
+        auto *ASI =
+            NewFBuilder.createAllocStack(InsertLoc, ArgDesc.Arg->getType());
+        ArgToAllocStackMap.insert(
+            std::pair<int, AllocStackInst *>(ArgDesc.Index, ASI));
+
+        auto *EAI = NewFBuilder.createInitExistentialAddr(
+            InsertLoc, ASI, NewArg->getType().getASTType(), NewArg->getType(),
+            Conformances);
+        /// If DestroyAddr is already there, then do not use [take].
+        NewFBuilder.createCopyAddr(
+            InsertLoc, NewArg, EAI,
+            ExistentialArgDescriptor[ArgDesc.Index].AccessType ==
+                    OpenedExistentialAccess::Immutable
+                ? IsTake_t::IsNotTake
+                : IsTake_t::IsTake,
+            IsInitialization_t::IsInitialization);
+        if (ExistentialArgDescriptor[ArgDesc.Index].DestroyAddrUse) {
+          NewFBuilder.createDestroyAddr(InsertLoc, NewArg);
+        } else {
+          MissingDestroyUse = true;
+        }
+        ValueMap.insert(std::make_pair(OrigArg, SILValue(ASI)));
+        assert(ValueMap.find(OrigArg) != ValueMap.end());
+        break;
+      }
+      case ExistentialRepresentation::Class: {
+        ///  Simple case: Create an init_existential.
+        /// %5 = init_existential_ref %0 : $T : $T, $P
+        auto *InitRef = NewFBuilder.createInitExistentialRef(
+            InsertLoc, ArgDesc.Arg->getType(), NewArg->getType().getASTType(),
+            NewArg, Conformances);
+        ValueMap.insert(std::make_pair(OrigArg, SILValue(InitRef)));
+        assert(ValueMap.find(OrigArg) != ValueMap.end());
+        break;
+      }
+      default: {
+        llvm_unreachable("Unhandled existential type in ExistentialTransform!");
+        break;
+      }
+      };
+    } else {
+      /// Arguments that are not rewritten.
+      auto Ty = params[ArgDesc.Index].getType();
+      auto MappedTy = ArgDesc.Arg->getType();
+      if (Ty->hasTypeParameter()) {
```

Comment:
You don't need the 'if'. It should either be fine to map it in unconditionally, or if that always fails, just don't map it in at all.